### PR TITLE
Openvino ep ort 5.1

### PR DESCRIPTION
**Pull request by sfatimar**
OpenVINO EP ORT 5.1 Branch
Changes for the new API to take in OpenVINO Provider Options
and compatibility with OV 2023.1

### Motivation and Context
The change is required for the new API to take in OpenVINO Provider Options
and make it seamless.

### CODE REVIEW

File: cmake/deps.txt [hunk 1]
```diff
@@ -41,4 +41,4 @@ safeint;https://github.com/dcleblanc/SafeInt/archive/ff15c6ada150a5018c5ef217240
 tensorboard;https://github.com/tensorflow/tensorboard/archive/373eb09e4c5d2b3cc2493f0949dc4be6b6a45e81.zip;67b833913605a4f3f499894ab11528a702c2b381
 cutlass;https://github.com/NVIDIA/cutlass/archive/refs/tags/v3.0.0.zip;0f95b3c1fc1bd1175c4a90b2c9e39074d1bccefd
 extensions;https://github.com/microsoft/onnxruntime-extensions/archive/94142d8391c9791ec71c38336436319a2d4ac7a0.zip;4365ac5140338b4cb75a39944a4be276e3829b3c
-eigen;https://gitlab.com/libeigen/eigen/-/archive/3.4/eigen-3.4.zip;ee201b07085203ea7bd8eb97cbcb31b07cfa3efb
\ No newline at end of file
+eigen;https://gitlab.com/libeigen/eigen/-/archive/3.4/eigen-3.4.zip;ee201b07085203ea7bd8eb97cbcb31b07cfa3efb
```
Comment (jywu-msft) [inline cmake/deps.txt:44]: this file wasn't actually modified right?

Comment (jywu-msft) [reply cmake/deps.txt:44]: if so, remove from change list.

**Comment by jywu-msft** [comment]
is cmake/external/emsdk supposed to be deleted?

File: onnxruntime/core/session/provider_registration.cc [hunk 2]
```diff
@@ -67,10 +68,17 @@ ORT_API_STATUS_IMPL(OrtApis::SessionOptionsAppendExecutionProvider,
   };
 
   if (strcmp(provider_name, "QNN") == 0) {
+  // if (strcmp(provider_name, "QNN") == 0) {
```
Comment (jywu-msft) [inline onnxruntime/core/session/provider_registration.cc:71]: this comment line seems accidental?

File: onnxruntime/core/session/provider_registration.cc [hunk 3]
```diff
@@ -115,7 +123,7 @@ ORT_API_STATUS_IMPL(OrtApis::SessionOptionsAppendExecutionProvider,
   } else {
     ORT_UNUSED_PARAMETER(options);
     status = OrtApis::CreateStatus(ORT_INVALID_ARGUMENT,
-                                   "Unknown provider name. Currently supported values are 'SNPE', 'XNNPACK', and 'AZURE'");
+                                   "Unknown provider name. Currently supported values are 'OPENVINO', 'SNPE', 'XNNPACK', and 'AZURE'");
```
Comment (jywu-msft) [inline onnxruntime/core/session/provider_registration.cc:126]: do you mind adding QNN and WEBNN since you are touching this line?

Comment (preetha-intel) [reply onnxruntime/core/session/provider_registration.cc]: Added QNN and WEBNN part of the Error status.

File: onnxruntime/core/session/provider_registration.cc (hunk 2 above)
Comment (preetha-intel) [reply onnxruntime/core/session/provider_registration.cc]: Yeah it was added accidentally. Removed now

**Comment by jywu-msft** [comment]
fyi, it says "This branch has conflicts that must be resolved"

File: onnxruntime/core/session/provider_registration.cc (hunk 3 above)
Comment (jywu-msft) [reply onnxruntime/core/session/provider_registration.cc]: thanks!

File: cmake/deps.txt (hunk 1 above)
Comment (sfatimar) [reply cmake/deps.txt]: We fixed this

**Comment by sfatimar** [comment]
@jywu-msft we have fixed conflicts with the branch

**Review by jywu-msft — APPROVED** [review]

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**